### PR TITLE
Adding participant removal by Button & confirm modal and by timeout.

### PR DESCRIPTION
**Pull request by tuhlmann**
This PR adds two features around participant removal:

1) There's a top centered button to kick our ALL participants from a room:

This opens a modal and when confirmed will remove all, active and inactive, users from this room and, if these are the only rooms they are in, will remove them from the players list as well.

**Caveat:**

The user that kicked off the removed all see the name modal again to enter a new name. Any other users still logged in will need to refresh their page to see the name modal. Please advice on how this could best be solved.

1) Everytime a player is updated, a lastSeen timestamp is renewed. If that timestamp is older than 24h, the user is removed from all rooms he's in and then from the players list as well.

**Caveat:**

There is currenly no board update if a use has been automatically been removed. Please advice on how to best add this.

### CODE REVIEW

**Comment by tuhlmann** [comment]
@mikepray Both issues are now fixed- any still open browser window will trigger the re-join by showing the name modal and then adds that new user to the board, no browser refresh necessary.

**Review by mikepray — CHANGES_REQUESTED** [review]
Looking great, did some testing and kicking participants works great. Some minor feedback in the changes requested.

How should the cleanup timeout work? Is it supposed to kick idle participants, or participants that have closed their socket connection after an hour or a day?

File: pointer-backend/src/services/room.service.ts [hunk 1]
```diff
@@ -45,11 +46,35 @@ export const deleteEstimates = (roomId: string): Boolean => {
                 player.estimate = "None";
             }
         });
+
         return true;
     }
     return false;
 }
 
+export const deleteParticipants = (roomId: string): [boolean, Player[] | undefined] => {
```
[the remark this answers is not in the document]
Comment (tuhlmann) [reply pointer-backend/src/services/room.service.ts:55]: Changed.

File: pointer-backend/src/services/room.service.ts [hunk 2]
```diff
@@ -67,4 +92,28 @@ export const getRoomState = (roomId: string) => {
...
     }
-}
\ No newline at end of file
+}
+
+/**
+ * Checks to see whether this player is present in any room
+ * @returns true if the player is present in any room, false otherwise.
+ */
+const playExistsInAnyRoom = (playerUid: string): boolean => {
```
[the remark this answers is not in the document]
Comment (tuhlmann) [reply pointer-backend/src/services/room.service.ts:101]: Fixed.

**Comment by tuhlmann** [comment]
@mikepray In player.service.ts I defined 2 timeouts:

```Typescript
```

**Comment by tuhlmann** [comment]
I implemented the requested changes, ready for another look.

**Review by mikepray — APPROVED** [review]

**Pull request merged**